R/filter_confirmation.R

```diff
@@ -0,0 +1,473 @@
+#' Filter Confirmed Observations
+#'
+#' @description
+#'
+#' The function filters observation using a condition taking subsequent
+#' observations into account. For example, it could select all observations with
+#' `AVALC == "Y"` and `AVALC == "Y"` for at least one subsequent observation.
+#' The input dataset is joined with itself to enable conditions taking variables
+#' from both the current observation and the subsequent observations into
+#' account. The suffix ".join" is added to the variables from the subsequent
+#' observations.
+#'
+#' In the oncology setting, for example, we use such processing to check if a
+#' response value can be confirmed by a subsequent assessment. This is commonly
+#' used in endpoints such as best overall response.
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified for `by_vars`, `join_vars`, and `order` are
+#'   expected.
+#'
+#' @param by_vars By variables
+#'
+#'   The specified variables are used as by variables for joining the input
+#'   dataset with itself.
+#'
+#' @param join_vars Variables to keep from joined dataset
+#'
+#'   The variables needed from the subsequent observations should be specified
+#'   for this parameter. The specified variables are added to the joined dataset
+#'   with suffix ".join". For example to select all observations with `AVALC ==
+#'   "Y"` and `AVALC == "Y"` for at least one subsequent visit `join_vars =
+#'   vars(AVALC, AVISITN)` and `filter = AVALC == "Y" & AVALC.join == "Y" &
+#'   AVISITN < AVISITN.join` could be specified.
+#'
+#'   The `*.join` variables are not included in the output dataset.
+#'
+#' @param first_cond Condition for selecting range of data
+#'
+#'   If this argument is specified, the subsequent observations are restricted
+#'   up to the first observation where the specified condition is fulfilled. If
+#'   the condition is not fulfilled for any of the subsequent observations, all
+#'   observations are removed.
+#'
+#' @param order Order
+#'
+#'   The observations are ordered by the specified order.
+#'
+#' @param filter Condition for selecting observations
+#'
+#'   The filter is applied to the joined dataset for selecting the confirmed
+#'   observations. The condition can include summary functions. The joined
+#'   dataset is grouped by the original observations. I.e., the summary function
+#'   are applied to all observations up to the confirmation observation. For
+#'   example, `filter = AVALC == "CR" & all(AVALC.join %in% c("CR", "NE")) &
```

Comment:
"For example _in the oncology setting when using this function for confirmed best overall response_, ...."

I think adding this frames this bit a bit better now or it would confuse non-Onco users

Reply:
Added